packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -426,9 +429,9 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
   void didUpdateWidget(SelectableRegion oldWidget) {
     super.didUpdateWidget(oldWidget);
     if (widget.focusNode != oldWidget.focusNode) {
-      oldWidget.focusNode.removeListener(_handleFocusChanged);
-      widget.focusNode.addListener(_handleFocusChanged);
-      if (widget.focusNode.hasFocus != oldWidget.focusNode.hasFocus) {
+      oldWidget.focusNode?.removeListener(_handleFocusChanged);
+      widget.focusNode?.addListener(_handleFocusChanged);
+      if (_focusNode.hasFocus != oldWidget.focusNode?.hasFocus) {
```

Comment:
If the widget goes from having null passed to focusNode to having a FocusNode passed, then _localFocusNode won't be disposed until SelectableRegion is disposed. Is that ok? Also if later it goes back to null again, then it will start reusing the old _localFocusNode, including possibly switching from focused to unfocused.

Would it be any better to set _localFocusNode only when it's needed and then dispose it and set it to null whenever it's not?

Reply:
It won't be a problem: the focus node won't be in the focus tree anymore if it's not used, because the `Focus` widget here will detach it when the new one is assigned, and if it's reattached, then where it is attached (and whether `FocusManager.instance.primaryFocus` is equal to that node) will determine whether it has focus, not any internal state.

You know, now that I look at it, I think that the `Focus` widget is being used incorrectly here anyhow.  It should be using `Focus.withExternalFocusNode` instead.  WIll fix that...